packages/twenty-server/src/engine/metadata-modules/relation-metadata/hooks/before-delete-one-relation.hook.ts

```diff
@@ -25,12 +30,12 @@ export class BeforeDeleteOneRelation implements BeforeDeleteOneHook {
       throw new UnauthorizedException();
     }
 
-    const relationMetadata =
-      await this.relationMetadataService.findOneWithinWorkspace(workspaceId, {
-        where: {
-          id: instance.id.toString(),
-        },
-      });
+    const relationMetadata = await this.relationMetadataRepository.findOne({
+      where: {
+        workspaceId,
+        id: instance.id.toString(),
+      },
```

Comment:
If you want to use repository, keep in mind that relations won't be loaded by default
```suggestion
      },
      relations: ['fromFieldMetadata', 'toFieldMetadata'],
```
otherwise it will fail below when retrieving the fields in relationMetadata.toFieldMetadata.isCustom for example